src/webgpu/api/operation/command_buffer/image_copy.spec.ts

```diff
@@ -1501,39 +1701,52 @@ aspect.
       mipLevel,
     } = t.params;
 
-    await t.selectDeviceOrSkipTestCase(kTextureFormatInfo[stencilFormat].feature);
+    await t.selectDeviceOrSkipTestCase(kTextureFormatInfo[format].feature);
 
-    const info = kTextureFormatInfo['stencil8'];
-    const copyWidth = copyWidthInBlocks * info.blockWidth;
-    const copyHeight = copyHeightInBlocks * info.blockHeight;
+    const bytesPerBlock = depthStencilFormatAspectSize(format, aspect);
     const rowsPerImage = copyHeightInBlocks + rowsPerImagePadding;
 
     const bytesPerRowAlignment = copyMethod === 'WriteTexture' ? 1 : kBytesPerRowAlignment;
     const bytesPerRow =
-      align(info.bytesPerBlock * copyWidthInBlocks, bytesPerRowAlignment) +
+      align(bytesPerBlock * copyWidthInBlocks, bytesPerRowAlignment) +
       bytesPerRowPadding * bytesPerRowAlignment;
 
-    const textureSize = [copyWidth << mipLevel, copyHeight << mipLevel, copyDepth] as const;
+    const textureSize = [
+      copyWidthInBlocks << mipLevel,
+      copyHeightInBlocks << mipLevel,
+      copyDepth,
+    ] as const;
     if (copyMethod === 'CopyT2B') {
-      await t.DoCopyFromStencilTest(
-        stencilFormat,
-        textureSize,
-        bytesPerRow,
-        rowsPerImage,
-        0,
-        mipLevel
-      );
+      if (aspect === 'depth-only') {
+        t.DoCopyTextureToBufferWithDepthAspectTest(
+          format,
+          [copyWidthInBlocks, copyHeightInBlocks, copyDepth] as const,
+          bytesPerRowPadding,
+          rowsPerImagePadding,
+          0,
+          0,
+          mipLevel
+        );
+      } else {
+        await t.DoCopyFromStencilTest(format, textureSize, bytesPerRow, rowsPerImage, 0, mipLevel);
+      }
     } else {
-      assert(copyMethod === 'CopyB2T' || copyMethod === 'WriteTexture');
+      assert(
+        aspect === 'stencil-only' && (copyMethod === 'CopyB2T' || copyMethod === 'WriteTexture')
+      );
       const initialDataSize = dataBytesForCopyOrFail({
         layout: { bytesPerRow, rowsPerImage },
         format: 'stencil8',
-        copySize: { width: copyWidth, height: copyHeight, depthOrArrayLayers: copyDepth },
+        copySize: {
+          width: copyWidthInBlocks,
+          height: copyHeightInBlocks,
+          depthOrArrayLayers: copyDepth,
+        },
```

Comment:
I believe dataBytesForCopyOrFail takes the size in pixels, not the size in blocks. (Anyway, there are no block-based depth/stencil formats and probably never will be, so it doesn't really matter.)
```suggestion
        copySize: textureSize,
```